src/System.Memory/src/System/Buffers/ReadOnlySequence_helpers.cs

```diff
@@ -11,288 +12,257 @@ namespace System.Buffers
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal bool TryGetBuffer(SequencePosition start, SequencePosition end, out ReadOnlyMemory<T> data, out SequencePosition next)
         {
-            if (start.Segment == null)
+            SequenceType type;
+            next = default;
+            int startIndex = 0;
+            int endIndex = 0;
+            object startSegment = start.Segment;
+
+            if (startSegment == null)
+            {
+                type = SequenceType.Empty;
```

Comment:
Why don't we return early for the null segment like we did before?

Reply:
Is it worth optimizing for empty segments?

Reply:
we still check the same thing `startSegment == null` but instead of just returning we do a bit more logic. Just makes things a bit more complicated. But I wouldn't optimize for it null segments, it is very rare.